File: project/src/main/java/com/moyeoba/project/api/kakao/KakaoProfileDto.java
package com.moyeoba.project.api.kakao;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.DateTimeException;
import java.time.LocalDate;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public class KakaoProfileDto {
    private String id;
    private String connected_at;
    private kakao_account kakao_account;

    //Filed Info Docs
    //https://developers.kakao.com/docs/latest/ko/kakaologin/rest-api#kakaoaccount
    @Data
    public class kakao_account {
        private String name;
        private String email;
    }
}

File: src/main/kotlin/com/moyeoba/moyeoba/api/kakao/KakaoProfileDto.kt
package com.moyeoba.moyeoba.api.kakao

import javax.xml.crypto.Data



data class KakaoProfileDto(
        var id: Long,
        var connected_at: String,
        var kakao_account: KakaoAccount
) {
    constructor() : this(0L, "", KakaoAccount())

    //Filed Info Docs
    //https://developers.kakao.com/docs/latest/ko/kakaologin/rest-api#kakaoaccount

     data class KakaoAccount (
            var name: String,
            var email: String
    ) {
        constructor(): this("", "")
    }
}
